src/test/fee_est/fee_est.cpp

```diff
@@ -0,0 +1,331 @@
+        } else if (match(arg, "odat", &value)) {
+            options.dataFileOut = value;
+        } else if (match(arg, "idat", &value)) {
+            options.dataFileIn = value;
+        } else if (match(arg, "dat", &value)) {
+            options.dataFileIn = options.dataFileOut = value;
+        } else if (match(arg, "cross")) {
+            options.cross = true;
+        } else if (match(arg, "help") || match(arg, "h") || match(arg, "?")) {
+            options.help = true;
+        } else if (arg[0] != '-' && options.logFileIn.empty()) {
+            options.logFileIn = arg;
+        } else {
+            fprintf(stderr, "Error: unexpected argument: '%s'\n", arg);
+            options.error = true;
+        }
+    }
+
+    if (options.logFileIn.empty() && !options.help) {
+        fprintf(stderr, "Error: missing required log file argument.\n");
+        options.error = true;
+    }
+
+    return options;
+}
+
+struct TxData {
+    TxData(CAmount fee, size_t size, int height) : fee(fee), size(size), height(height) {}
+    CAmount fee;
+    size_t size;
+    int height;
+    int expectedBlocks = -1;
+    int actualBlocks = -1;
+};
+
+using TxMap = std::map<uint256, TxData>;
+
+const char* const GRAPH_HTML = R"(<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Fee Graph</title>
+<script src="https://d3js.org/d3.v3.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/vega/3.0.0-beta.30/vega.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/vega-lite/2.0.0-beta.2/vega-lite.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/vega-embed/3.0.0-beta.14/vega-embed.js"></script>
+<style media="screen">
+.vega-actions a {
+    margin-right: 5px;
+}
+</style>
+<div id=vis></div>
+<script>
+var spec = {
+  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
+  "width": 800,
+  "height": 600,
+  "layer": [
+    {
+        "data": {
+            "values": %s
+        },
+        "mark": "tick",
+        "encoding": {
+            "x": {"field": "blocks", "type": "quantitative"},
+            "y": {"field": "feeRate", "type": "quantitative"},
+            "color": {"field": "height", "type": "quantitative"}
+        }
+    },
+    {
+      "data": {
+        "values": %s
+      },
+      "encoding": {
+        "x": {"field": "blocks", "type": "quantitative"},
+        "y": {"field": "feeRate", "type": "quantitative"},
+        "color": {"field": "mode", "type": "nominal"}
+      },
+      "mark": "line",
+    }
+  ]
+}
+vega.embed("#vis", spec);
+</script>
+)";
+
+// Maximum number of randomly sampled transactions to show on graph.
+const size_t SAMPLE_TXS = 50000;
+
+// Drop 1/20th of sampled transactions with highest feerates if they are
+// outliers that would change the scale of the graph.
+const int DROP_FRAC = 20;
+
+void WriteGraph(const std::string& filename,
+    const TxMap& txMap,
+    const std::vector<unsigned int>& targets,
+    CBlockPolicyEstimator& estimator)
```

Comment:
538c780d078ecf400eb7b509f9fbcd176cdd290c
```suggestion
    const CBlockPolicyEstimator& estimator)
```